pandas/core/resample.py

```diff
@@ -856,10 +856,32 @@ def size(self):
         # It's a special case as higher level does return
         # a copy of 0-len objects. GH14962
         result = self._downsample("size")
-        if not len(self.ax) and isinstance(self._selected_obj, ABCDataFrame):
+        if not len(self.ax):
+            if isinstance(self._selected_obj, ABCDataFrame):
+                from pandas import Series
+
+                result = Series([], index=result.index, dtype="int64")
+            elif isinstance(self._selected_obj, ABCSeries):
+                from pandas import Series
+
+                result = Series(
+                    [], index=result.index, dtype="int64", name=self._selected_obj.name
+                )
+            else:
+                raise AssertionError("Unexpected type result")
```

Comment:
Is this actually hit? Ideally we wouldn't have unreachable code sections.

If you need / want to assert that we only have a frame or series here, you can make that as an `assert`

Reply:
And perhaps we can simplify this whole thing to something like

```python
if not len(self.ax):
    if self._selected_obj.ndim == 1:
        name = self._selected_obj.name
    else:
        name = None
    return Series([], index=result.index, name=name, dtype="int64")
```
```

Reply:
Yes, this should not be hit AFAIU. I guess I can just test `isinstance(self._selected_obj, ABCSeries)` anyway.